src/test/java/com/myThreadPool/MyThreadPoolTests.java

```diff
@@ -3,27 +3,30 @@
 import com.myThreadPool.bankAccountApp.AddOperation;
 import com.myThreadPool.bankAccountApp.CustomBankAccount;
 import org.junit.Assert;
-import org.junit.Test;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
 
-public class MyThreadPoolTests {
+import java.util.ArrayList;
+import java.util.List;
 
+@Listeners (TestListener.class)
+public class MyThreadPoolTests {
 
-    @Test
-    public void addToTheSameAccountTenTimesUsingFiveThreads() throws InterruptedException {
-        MyThreadPool threadPool = new MyThreadPool(3,5);
+    @Test (dataProvider = "queueSize-noOfThreads", dataProviderClass = DataProviders.class)
+    public void addToTheSameAccountMultipleTimes (ThreadPoolTestData testData) throws InterruptedException {
+        MyThreadPool threadPool = new MyThreadPool(testData.getQueueSize(), testData.getNoOfThreads());
         CustomBankAccount account = new CustomBankAccount();
+        List<AddOperation> listOfOperations = new ArrayList<>();
 
-        for(int taskNumber = 1 ; taskNumber <= 10; taskNumber++) {
+        for(int taskNumber = 1 ; taskNumber <= testData.getNoOfTasks(); taskNumber++) {
             AddOperation task = new AddOperation(account);
-            threadPool.submit(task, 10000);
+            threadPool.submit(task, testData.getDelay());
+            listOfOperations.add(task);
         }
 
-        double expectedResult = 0;
-        for (int i = 1; i <= 10; i++) {
-            expectedResult += 100;
-            expectedResult += (expectedResult * 0.01);
-        }
-        Thread.sleep(30000);
+
+        double expectedResult = BankOperationHelper.calculateExpectedBalance(listOfOperations);
+        Thread.sleep(60000);
         Assert.assertEquals("The final account balance is not correct", expectedResult, account.getBalance(), 0.00001);
```

Comment:
In order to keep the tests well-organised, try to use one TestNG or Junit in the same time, not both.
This `Assert` is from`org.junit` package whereas `@Test` is from `org.testng.annotations`.
For `Assert` I suggest to use `org.testng` and skip using junit